dipy/core/gradients.py

```diff
@@ -67,6 +69,71 @@ def __init__(self, gradients, big_delta=None, small_delta=None,
         self.big_delta = big_delta
         self.small_delta = small_delta
         self.b0_threshold = b0_threshold
+        if isinstance(btens, str):
+            b_tensors = np.zeros((len(self.bvals), 3, 3))
+            if btens == 'LTE':
```

Comment:
We may gain some consistency and it may be easier to maintain if the supported tensor types are put to an enum, e.g.

```
class TensorEncoding(enum.Enum):
    """Tensor encoding types."""
    LTE = 'linear_tensor_encoding'
    PTE = 'planar_tensor_encoding'
    STE = 'spherical_tensor_encoding'
```

and then check as e.g. `if btens == TensorEncoding.LTE`. The `ValueError` exception below will need to be modified to use the above enum instead of the strings. Similarly, the test will need to be updated if the enum is adopted.

Given that the `LTE`, `PTE`, `STE` acronyms have been adopted in literature, may be having `TensorEncoding.LTE` is a bit verbose, so the enum may need another name.